vumi/middleware/tests/test_session_length.py

```diff
@@ -0,0 +1,141 @@
+
+from vumi.message import TransportUserMessage
+from vumi.middleware.session_length import SessionLengthMiddleware
+from vumi.tests.helpers import VumiTestCase, PersistenceHelper
+
+SESSION_NEW, SESSION_CLOSE = (
+    TransportUserMessage.SESSION_NEW, TransportUserMessage.SESSION_CLOSE)
+
+
+class TestStaticProviderSettingMiddleware(VumiTestCase):
+    def setUp(self):
+        self.persistence_helper = self.add_helper(PersistenceHelper())
+
+    @inlineCallbacks
+    def mk_middleware(self, config={}):
+        dummy_worker = object()
+        config = self.persistence_helper.mk_config(config)
+        mw = SessionLengthMiddleware(
+            "static_provider_setter", config, dummy_worker)
+        yield mw.setup_middleware()
+        self.redis = mw.redis
+        self.add_cleanup(mw.teardown_middleware)
+        returnValue(mw)
+
+    def mk_msg(self, to_addr, from_addr, session_event=SESSION_NEW):
+        msg = TransportUserMessage(
+            to_addr=to_addr, from_addr=from_addr,
+            transport_name="dummy_connector",
+            transport_type="dummy_transport_type",
+            session_event=session_event)
+        return msg
+
+    @inlineCallbacks
+    def test_incoming_message_session_start(self):
+        mw = yield self.mk_middleware()
+        msg_start = self.mk_msg('+12345', '+54321')
+
+        msg = yield mw.handle_inbound(msg_start, "dummy_connector")
+        value = yield self.redis.get('+54321:session_created')
+        self.assertTrue(value is not None)
+        self.assertTrue(isinstance(
+                msg['helper_metadata']['session']['session_start'], float))
+
+    @inlineCallbacks
+    def test_incoming_message_session_end(self):
+        mw = yield self.mk_middleware()
+        msg_end = self.mk_msg('+12345', '+54321', session_event=SESSION_CLOSE)
+
+        msg = yield mw.handle_inbound(msg_end, "dummy_connector")
+        self.assertTrue(isinstance(
+                msg['helper_metadata']['session']['session_end'], float))
+
+    @inlineCallbacks
+    def test_incoming_message_session_start_end(self):
+        mw = yield self.mk_middleware()
+        msg_start = self.mk_msg('+12345', '+54321')
+        msg_end = self.mk_msg('+12345', '+54321', session_event=SESSION_CLOSE)
+
+        msg = yield mw.handle_inbound(msg_start, "dummy_connector")
+        value = yield self.redis.get('+54321:session_created')
+        self.assertTrue(value is not None)
+        self.assertTrue(isinstance(
+                msg['helper_metadata']['session']['session_start'], float))
+
+        msg = yield mw.handle_inbound(msg_end, "dummy_connector")
+        value = yield self.redis.get('+54321:session_created')
+        self.assertTrue(value is None)
+        self.assertTrue(isinstance(
+                msg['helper_metadata']['session']['session_start'], float))
+        self.assertTrue(isinstance(
+            msg['helper_metadata']['session']['session_end'], float))
```

Comment:
I think it might be worth stubbing `time.time` with `self.patch` to check that we are storing the right values for the start and end.